References: bsc#1174176, bsc#1174139
Current libxl does not support adding a floppy. Code is included
in this patch anyways to correctly set up a bus if a floppy were
added.


Index: virt-manager-4.2.0/virtManager/addhardware.py
===================================================================
--- virt-manager-4.2.0.orig/virtManager/addhardware.py
+++ virt-manager-4.2.0/virtManager/addhardware.py
@@ -489,6 +489,9 @@ class vmmAddHardware(vmmGObjectUI):
             buses = domcaps.devices.disk.get_enum("bus").get_values()
         else:
             buses = vmmAddHardware.disk_old_recommended_buses(guest)
+        # libxl doesn't support floppy but request one from qemu anyways if specified
+        if guest.conn.is_xen() and devtype == "floppy":
+            buses.append("fdc")
 
         bus_map = {
             "disk": ["ide", "sata", "scsi", "sd", "usb", "virtio", "xen"],
@@ -508,6 +511,7 @@ class vmmAddHardware(vmmGObjectUI):
             "usb": _("USB"),
             "virtio": _("VirtIO"),
             "xen": _("Xen"),
+            "fdc": _("FDC"),
         }
         return bus_mappings.get(bus, bus)
 
@@ -665,8 +669,9 @@ class vmmAddHardware(vmmGObjectUI):
                       "drive-harddisk", _("Disk device")])
         target_model.append([DeviceDisk.DEVICE_CDROM,
                       "media-optical", _("CDROM device")])
-        target_model.append([DeviceDisk.DEVICE_FLOPPY,
-                      "media-floppy", _("Floppy device")])
+        if not self.conn.is_xen():
+            target_model.append([DeviceDisk.DEVICE_FLOPPY,
+                          "media-floppy", _("Floppy device")])
         if self.conn.is_qemu() or self.conn.is_test():
             target_model.append([DeviceDisk.DEVICE_LUN,
                           "drive-harddisk", _("LUN Passthrough")])
